core/src/scene/scene.h

```diff
@@ -107,6 +108,8 @@ class Scene {
 
     std::shared_ptr<Texture> getTexture(const std::string& name) const;
 
+    std::atomic_ushort m_resourceLoad;
```

Comment:
I think any operations on word-sized variables are guaranteed to be atomic without any extra guards (on ARM at least, need to check for x86).